### PR TITLE
@grafana/toolkit: improve the circleci task

### PR DESCRIPTION
**Pull request by ryantxu**
This PR makes some minor improvements to the circle task

1. Adds build info to plugin.json
2. adds dependencies to deployed artifacts
3. Makes sure prettier has content before writing (avoid writing empty files)
4. renames 'bundle' to 'package' and saves stuff in a 'packages' file

### CODE REVIEW

**Comment by ryantxu** [comment]
@dprokop Trying to test this, I get stuck running:
```
yarn packages:prepare
```

I see an error about:
```
Running "exec:typecheckRoot" (exec) task
$ tsc --noEmit
public/app/core/components/Animations/SlideDown.tsx:44:13 - error TS2322: Type '{ inProp: boolean; alignContent?: string; alignItems?: string; alignSelf?: string; animationDelay?: string; animationDirection?: string; animationDuration?: string; animationFillMode?: string; animationIterationCount?: ReactText; ... 734 more ...; vectorEffect?: VectorEffectProperty; }' is not assignable to type 'CSSProperties'.
  Object literal may only specify known properties, and 'inProp' does not exist in type 'CSSProperties'.

44             inProp,
               ~~~~~~

  node_modules/@types/react/index.d.ts:1657:9
    1657         style?: CSSProperties;
                 ~~~~~
    The expected type comes from property 'style' which is declared here on type 'DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>'

public/app/features/explore/AdHocFilterField.tsx:37:3 - error TS2416: Property 'componentDidUpdate' in type 'AdHocFilterField<TQuery, TOptions>' is not assignable to the same property in base type 'PureComponent<Props<TQuery, TOptions>, State, any>'.
  Type '(prevProps: Props<DataQuery, DataSourceJsonData>) => void' is not assignable to type '(prevProps: Readonly<Props<TQuery, TOptions>>, prevState: Readonly<State>, snapshot?: any) => void'.
    Types of parameters 'prevProps' and 'prevProps' are incompatible.
      Type 'Readonly<Props<TQuery, TOptions>>' is not assignable to type 'Props<DataQuery, DataSourceJsonData>'.
        Types of property 'datasource' are incompatible.
          Type 'DataSourceApi<TQuery, TOptions>' is not assignable to type 'DataSourceApi<DataQuery, DataSourceJsonData>'.
            Types of property 'components' are incompatible.
              Type 'DataSourcePluginComponents<DataSourceApi<TQuery, TOptions>, TQuery, TOptions>' is not assignable to type 'DataSourcePluginComponents<DataSourceApi<DataQuery, DataSourceJsonData>, DataQuery, DataSourceJsonData>'.
                Types of property 'QueryEditor' are incompatible.
                  Type 'ComponentType<QueryEditorProps<DataSourceApi<TQuery, TOptions>, TQuery, TOptions>>' is not assignable to type 'ComponentType<QueryEditorProps<DataSourceApi<DataQuery, DataSourceJsonData>, DataQuery, DataSourceJsonData>>'.
                    Type 'ComponentClass<QueryEditorProps<DataSourceApi<TQuery, TOptions>, TQuery, TOptions>, any>' is not assignable to type 'ComponentType<QueryEditorProps<DataSourceApi<DataQuery, DataSourceJsonData>, DataQuery, DataSourceJsonData>>'.
                      Type 'ComponentClass<QueryEditorProps<DataSourceApi<TQuery, TOptions>, TQuery, TOptions>, any>' is not assignable to type 'ComponentClass<QueryEditorProps<DataSourceApi<DataQuery, DataSourceJsonData>, DataQuery, DataSourceJsonData>, any>'.
                        Types of parameters 'props' and 'props' are incompatible.
                          Type 'QueryEditorProps<DataSourceApi<DataQuery, DataSourceJsonData>, DataQuery, DataSourceJsonData>' is not assignable to type 'QueryEditorProps<DataSourceApi<TQuery, TOptions>, TQuery, TOptions>'.

37   componentDidUpdate(prevProps: Props) {
     ~~~~~~~~~~~~~~~~~~

public/app/features/plugins/PluginPage.tsx:149:31 - error TS2326: Types of property 'type' are incompatible.
  Type '"dashboard"' is not assignable to type 'PluginIncludeType'.

149       if (find(meta.includes, { type: 'dashboard' })) {
...
``

File: packages/grafana-toolkit/tsconfig.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "extends": "../tsconfig.json",
-  "include": ["src/**/*.ts"],
+  "include": ["src/**/*.ts", "../../public/app/types/jquery/*.ts"],
```
Comment (ryantxu) [inline packages/grafana-toolkit/tsconfig.json]: should we just move the jquery types to toolkit?

File: packages/grafana-toolkit/src/cli/tasks/package.build.ts [hunk 2]
```diff
@@ -43,6 +43,21 @@ export const savePackage = useSpinner<SavePackageOptions>(
...
+  if (name.endsWith('/ui')) {
+    deps['@grafana/data'] = version;
+  } else if (name.endsWith('/runtime')) {
+    deps['@grafana/data'] = version;
+    deps['@grafana/ui'] = version;
+  } else if (name.endsWith('/toolkit')) {
+    deps['@grafana/data'] = version;
+    deps['@grafana/ui'] = version;
+  }
+
```
Comment (ryantxu) [inline packages/grafana-toolkit/src/cli/tasks/package.build.ts]: @dprokop -- thoughts on this approach?   Is this the right place?

Comment (dprokop) [reply packages/grafana-toolkit/src/cli/tasks/package.build.ts]: I think it's fine as it's saved directly to dist and the original package.json ain't modified.

**Comment by ryantxu** [comment]
@dprokop -- this is ready to merge (I think).

It generates a report like this:
https://373-192459187-gh.circle-artifacts.com/0/home/circleci/plugin/ci/report.json

File: packages/grafana-toolkit/tsconfig.json (hunk 1 above)
Comment (dprokop) [reply packages/grafana-toolkit/tsconfig.json]: Nah, I don't think so. We would have to move those to grafana/data as well as it uses them too. I feel referencing those from core is not that bad here especially those are needed only on compilation time which happens in core env anyways.

File: packages/grafana-toolkit/src/cli/tasks/plugin.build.ts [hunk 3]
```diff
@@ -82,12 +82,19 @@ export const prettierCheckPlugin = useSpinner<Fixable>('Prettier check', async (
...
+              fs.writeFile(s, fixed, err => {
+                if (err) {
+                  console.log('Error fixing ' + s, err);
+                  failed = true;
+                } else {
+                  console.log('Fixed: ' + s);
+                }
+              });
+            } else {
+              console.log('No automatic fix for: ' + s);
```
Comment (dprokop) [inline packages/grafana-toolkit/src/cli/tasks/plugin.build.ts]: 👍

**Pull request merged**